java/grpc-client/src/main/java/io/vitess/client/grpc/GrpcClient.java

```diff
@@ -81,23 +81,36 @@
  * GrpcClient is a gRPC-based implementation of Vitess RpcClient.
  */
 public class GrpcClient implements RpcClient {
+  private static final Duration DEFAULT_TIMEOUT = Duration.standardSeconds(30);
+
   private final ManagedChannel channel;
   private final String channelId;
   private final VitessStub asyncStub;
   private final VitessFutureStub futureStub;
+  private final Duration timeout;
 
   public GrpcClient(ManagedChannel channel) {
     this.channel = channel;
     channelId = toChannelId(channel);
     asyncStub = VitessGrpc.newStub(channel);
     futureStub = VitessGrpc.newFutureStub(channel);
+    timeout = DEFAULT_TIMEOUT;
+  }
+
+  public GrpcClient(ManagedChannel channel, Context context) {
+    this.channel = channel;
+    channelId = toChannelId(channel);
+    asyncStub = VitessGrpc.newStub(channel);
+    futureStub = VitessGrpc.newFutureStub(channel);
+    timeout = context.getTimeout() != null ? context.getTimeout() : DEFAULT_TIMEOUT;
   }
 
-  public GrpcClient(ManagedChannel channel, CallCredentials credentials) {
+  public GrpcClient(ManagedChannel channel, CallCredentials credentials, Context context) {
     this.channel = channel;
     channelId = toChannelId(channel);
     asyncStub = VitessGrpc.newStub(channel).withCallCredentials(credentials);
     futureStub = VitessGrpc.newFutureStub(channel).withCallCredentials(credentials);
+    timeout = context.getTimeout() != null ? context.getTimeout() : DEFAULT_TIMEOUT;
```

Comment:
Also check that context is a positive number.

Reply:
👍 updated